### PR TITLE
Add a generic HTTPPost node

### LINKED ISSUE #1330: Generic HTTP Post node
While the HTTPOut node is useful for making node data available as a REST endpoint, it would also be useful to be able to POST the data to an HTTP endpoint, much like how the alert service allows you to POST alerts to a generic HTTP endpoint.

This is especially useful for publishing kapacitor statistics to an HTTP endpoint (like logstash) that can then format and push the data to some backend database.  We have an internal proprietary bus for metric data and want this integration to be able to consume kapacitor statistics in to our metrics database.

### PR DESCRIPTION
**Pull request by sputnik13**
An HTTPPost node to mimic the behavior of HTTPOut, with the difference
that data is POST'd to an HTTP endpoint rather than made available at a
kapacitor endpoint.

Closes #1330

Tested to ensure data is POST'd properly to a logstash instance, from which
metric data was pushed through our metric pipeline to a time series database.

Integration tests for stream and batch modes added and verified.

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

###### Required only if applicable
_You can erase any checkboxes below this note if they are not applicable to your Pull Request._
- [x] [TICKscript Spec](https://github.com/influxdata/kapacitor/blob/master/tick/TICKscript.md) updated

### CODE REVIEW

**Comment by nathanielc** [comment]
@sputnik13 On first read through this look great! :+1: 

Questions:

* Do you need support for authentication? Meaning using basic auth, to authenticate against the target.
* Do you need the ability to add headers to the POST request?

I ask because both of these have been requested and are being added in #1254 

@desa Seems like we can use your work in #1254 to add the same features to this new `httpPost` node. Thoughts?

**Comment by desa** [comment]
@nathanielc yeah I think that makes sense.

**Comment by nathanielc** [comment]
@desa Yeah, I think it makes sense to merge this one first. Thanks

File: pipeline/http_post.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// An HTTPPostNode will take the incoming data stream and POST it to an HTTP endpoint.
+//
+// Example:
+//    stream
+//        |window()
+//            .period(10s)
+//            .every(5s)
+//        |top('value', 10)
+//        //Post the top 10 results over the last 10s updated every 5s.
+//        |httpPost('http://example.com/api/top10')
```
Comment (nathanielc) [inline pipeline/http_post.go]: @desa Does this TICKscript usage work well for how your post service is going to work? Is there a change that would make is easier more consistent with your changes?

Comment (desa) [inline pipeline/http_post.go]: having an optional property method called `endpoint` would be useful, but I can add that once this is merged.

**Comment by sputnik13** [comment]
@nathanielc sounds good, except during further internal testing we saw kapacitor hard crash when the POST endpoint wasn't available...  not sure that's the best behavior, so I need to look in to this a bit more

**Comment by sputnik13** [comment]
@nathanielc it seems like closing resp.Body when http.Post fails causes a hard crash

**Comment by desa** [comment]
@sputnik13 is this PR ready to go? If so, could you rebase onto master?

**Comment by desa** [comment]
@sputnik13 From what I can tell this looks ready to go. I'm going to rebase and merge this PR.

**Comment by desa** [comment]
@sputnik13 is this PR ready to go? If so, could you squash and rebase onto master?

**Comment by sputnik13** [comment]
@desa I'm sorry, I was busy with work, hadn't had a chance to check back, thanks for merging it

**Comment by desa** [comment]
@sputnik13 no worries 👍 . Thanks for the PR.

**Pull request merged**